yaas/host.py

```diff
@@ -1,4 +1,87 @@
 # coding: utf-8
 
+from __future__ import absolute_import
+from __future__ import print_function
+
+import inspect
+import pprint
+import requests
+
+from . import config
+
+
 def command(parser, args):
-    pass
+    """ Interact with registered agents. """
+    subparsers = parser.add_subparsers(dest='subcommand')
+    subcommands = {
+        'list': _list,
+        'show': _show
+        }
+    for name, fn in subcommands.items():
+        subparsers.add_parser(name, help=inspect.getdoc(fn))
+    subargs, extra = parser.parse_known_args(args)
+    subcommands[subargs.subcommand](
+        subparsers.choices[subargs.subcommand],
+        extra)
+
+
+def _list(parser, args):
+    """ List all registered agents. """
+    response = requests.get(
+        config.href('/api/v1/hosts'),
+        **config.requests_opts())
+    if config.args.raw:
+        pprint.pprint(response.json())
+    else:
+        for item in response.json()['items']:
+            print(item['Hosts']['host_name'])
+
+
+def _show(parser, args):
+    """ Show registered agent info. """
+    parser.add_argument(
+        '--hardware',
+        action='store_true',
+        default=False,
+        help="Print hardware specs.")
+    subargs, extra = parser.parse_known_args(args)
+    for host_name in extra:
+        response = requests.get(
+            config.href('/api/v1/hosts/{host}'.format(host=host_name)),
+            **config.requests_opts())
+        if config.args.raw:
```

Comment:
This commit alone allows `HTTPError`s to bubble up to the user, but #13 will fix that.